File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.js
function gcd(a, b) {
    if (b === 0) {
        return a;
    }
    return gcd(b, a % b);
}


function solution(numer1, denom1, numer2, denom2) {
    const numer = (numer1 * denom2) + (numer2 * denom1);
    const denom = denom1 * denom2;
    
    const gcdValue = gcd(numer, denom);
    
    return [
        Math.trunc(numer / gcdValue),
        Math.trunc(denom / gcdValue)
    ];
}

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.py
import math

def solution(numer1, denom1, numer2, denom2):
    new_number = (numer1 * denom2) + (numer2 * denom1)
    new_denom = denom1 * denom2
    
    gcd_value = math.gcd(new_number, new_denom)
    
    return [new_number // gcd_value, new_denom // gcd_value]